qt/widgets/common/src/MWRunFiles.cpp

```diff
@@ -510,65 +511,101 @@ void MWRunFiles::clear() {
   m_uiForm.fileEditor->setText("");
 }
 
+/**
+* Finds the files if the user has changed the parameter text.
+*/
+void MWRunFiles::findFiles() { findFiles(m_uiForm.fileEditor->isModified()); }
+
 /**
 * Finds the files specified by the user in a background thread.
 */
-void MWRunFiles::findFiles() {
+void MWRunFiles::findFiles(bool isModified) {
+  auto searchText = m_uiForm.fileEditor->text();
+  const auto directory = findFilesDirectory(searchText);
+
   // Set the values for the thread, and start it running.
-  QString searchText = m_uiForm.fileEditor->text();
+  if (directory)
+    m_foundFiles.append(directory.get());
+  else {
+    if (isModified) {
+      // Reset modified flag.
+      m_uiForm.fileEditor->setModified(false);
+      searchText = findFilesGetSearchText(searchText);
+    }
+    runFindFiles(searchText);
```

Comment:
Shouldn't this be inside the `isModified` check or it's going to run even if the text in the box hasn't changed?